### ISSUE TITLE
sveltekit:refresh

### ISSUE THREAD

**Issue by Rich-Harris** [sveltejs/kit#546]
At present you can force an `<a>` tag to trigger a full page reload on click by adding either `target="_self"` or `rel="external"`.

Neither is particularly explicit. They cause a reload because &lt;handwave&gt;that's how SPA routers work&lt;/handwave&gt;, but in the name of clearly communicating intent and having a useful thing to add to the docs, it might be good to add a `sveltekit:refresh` attribute alongside `sveltekit:prefetch` and `sveltekit:noscroll`. 

It's duplicative, but maybe that's not the worst thing in the world?

**Comment by Conduitry** [comment]
Is the intent here that you would need `sveltekit:refresh` to make a link do regular navigation, even if it had `target` or `rel=external` or `download`? Or, what would the rules for this now look like?

I'm also not sold on `sveltekit:refresh` as a name - it doesn't really seem to convey exactly what we want to convey, but I'm not sure how to convey that succinctly. 'Refresh' means to me more a refresh of the same page you're already on.

**Comment by Rich-Harris** [comment]
> Or, what would the rules for this now look like?

`rel="external"`, `target="_self"` and `sveltekit:refresh` would all cause the router not to handle the navigation; the first two out of deference to established conventions, the latter in the name of clarity.

Though you make a good point that `refresh` doesn't exactly provide the clarity we seek. Very much open to suggestions (which could include 'just leave it as it is')

**Comment by Conduitry** [comment]
`download` should also defer to the browser's handling - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attributes - If we're inheriting Sapper's currently handling (and page.js's) this should already be the case.

I'll try to think about a better name for this. I did come up empty while writing my last comment, but I didn't think about it for very long.

**Comment by Rich-Harris** [comment]
`sveltekit:ignore` could work, as in 'SvelteKit's router should ignore this'

**Comment by Conduitry** [comment]
I think I'm currently leaning toward `sveltekit:no[something]` but don't know what the something should be.

**Comment by benmccann** [comment]
`sveltekit:noroute` perhaps?

**Comment by mquandalle** [comment]
Maybe `reload`? It's similar to `refresh` but implicitly applies to the “app” instead of the “page” — reload the app, refresh the page.

**Comment by PuruVJ** [comment]
+1 to @mquandalle's suggestion, this seems like the best naming for this

**Comment by Kapsonfire-DE** [comment]
while i was thinking about reload as well - i think "fullreload" would be better - why?


gonna start:
noroute feels wrong - its still a route but it should not use the router for navigating and it does not fullfill the need for its usage

sveltekit:ignore - yeah this fits better.. sveltekit should ignore this link - its more versatile for different use cases

sveltekit:reload - yeah but what does it reload. just the current route, all layout files... well and here we go... "fullreload" is what really happens

**Comment by simonsarris** [comment]
FYI to people finding this via search, you probably now want to read [`data-sveltekit-reload`](https://kit.svelte.dev/docs/link-options#data-sveltekit-reload)

**Comment by oodavid** [comment]
This has been seriously tripping me up recently, to the point I nearly abandoned kit for svelte-spa-router. I've spent the past part of 2 days trying to figure out why every link causes an entire refresh.

TLDR; I'd been applying `target="_self"` to my links by default as it matches the [defaults for the spec](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#target) (sometimes I like to be explicit).

Personally, I think this isn't expected behaviour, after all, why would setting an attribute to it's default value be any different to not setting the attribute in the first place?
